src/rules/extensions.js

```diff
@@ -147,24 +149,49 @@ module.exports = {
         const extensionForbidden = isUseOfExtensionForbidden(extension)
         if (extensionRequired && !extensionForbidden) {
           context.report({
-            node: source,
+            node,
             message:
               `Missing file extension ${extension ? `"${extension}" ` : ''}for "${importPath}"`,
           })
         }
       } else if (extension) {
         if (isUseOfExtensionForbidden(extension) && isResolvableWithoutExtension(importPath)) {
           context.report({
-            node: source,
+            node,
             message: `Unexpected use of file extension "${extension}" for "${importPath}"`,
           })
         }
       }
     }
 
-    return {
-      ImportDeclaration: checkFileExtension,
-      ExportNamedDeclaration: checkFileExtension,
+    function checkImportFileExtension(node) {
+      const { source } = node
+
+      // bail if the declaration doesn't have a source, e.g. "export { foo };"
+      if (!source) return
+
+      const importPath = source.value
+
+      checkFileExtension(importPath, source)
+    }
+
+    function checkCommonJSFileExtension(node) {
+      if (!isStaticRequire(node)) return
+
+      const importPath = node.arguments[0].value
+
+      checkFileExtension(importPath, node)
+    }
+
+    const ext = {
+      ImportDeclaration: checkImportFileExtension,
+      ExportNamedDeclaration: checkImportFileExtension,
     }
+
+    if (props.commonjs) {
+      ext.CallExpression = checkCommonJSFileExtension
```

Comment:
def would prefer avoiding mutation here and using an inline undefined, if it would work

Reply:
Changed to inline.

Reply:
After changing the line to 

```
CallExpression: props.commonjs ? checkCommonJSFileExtension : undefined,
```

I see CI broke with the following error (https://travis-ci.org/benmosher/eslint-plugin-import/jobs/524246160):

```
1) extensions valid import a from "a":
     TypeError [ERR_INVALID_ARG_TYPE]: Error while loading rule 'extensions': The "listener" argument must be of type Function. Received type undefined
      at Object.keys.forEach.selector (node_modules/eslint/lib/eslint.js:873:29)
      at Array.forEach (<anonymous>)
      at Object.keys.filter.forEach.key (node_modules/eslint/lib/eslint.js:872:39)
      at Array.forEach (<anonymous>)
      at EventEmitter.module.exports.api.verify (node_modules/eslint/lib/eslint.js:841:93)
      at runRuleForItem (node_modules/eslint/lib/testers/rule-tester.js:387:38)
      at testValidTemplate (node_modules/eslint/lib/testers/rule-tester.js:420:28)
      at Context.RuleTester.it (node_modules/eslint/lib/testers/rule-tester.js:548:25)
```

Reply:
gotcha, i guess this approach is fine then.

Reply:
Reverted and pushed.